### ISSUE TITLE
Unable to reproduce the OpenBind-0 RnP2025-preview results end to end from the released inputs

### ISSUE THREAD

**Issue by HBioquant** [aqlaboratory/openfold-3#413]
Hi OpenFold-3 team,

First, thank you for your work on advancing the OpenFold3 open-source ecosystem and for releasing the OpenBind data, model weights, and benchmarking results. This is excellent work and provides a valuable foundation for further development by the community.

I am attempting an independent, end-to-end reproduction of the OpenBind-0 results on RnP2025-preview:
 - Dataset: RnP2025-preview on Zenodo (https://zenodo.org/records/22037460)
 - Released benchmark results: OpenBind-0 benchmarking results (https://github.com/OpenBind-Consortium/OpenBind-0-model-release-info/tree/main/benchmarking_results)

To my understanding, this is the latest available RnP benchmark from the pending PLINDER 2026 update and is suitable for evaluating protein–small-molecule cofolding models with a training cutoff no later than June 30, 2025.

I can reproduce Figure 1 (https://openbind.uk/news/blog-openbind-0-advancing-open-molecular-structure-prediction) by aggregating the released ”per_interface_breakdown.parquet“. However, when I rerun inference from the released Zenodo queries, MSAs, templates, and official checkpoints, the resulting OpenBind-0 success rates do not match the released results **(OpenBind-0 without templates vs reported results: Top-1, −9.04 %; Oracle, −5.94 %; OpenBind-0 with templates vs reported results: Top-1, −6.20 %; Oracle, −5.68 %).** I also cannot exactly reconstruct the populations and results used for Figures 2–5 from the currently available artifacts. I'm sorry, for some reason I cannot directly provide my complete reproduction image.

Thus, the plotting and aggregation logic can reproduce Figure 1 from the released parquet, but rerunning the released checkpoint from the public inputs does not reproduce the OpenBind-0 values.

----
**_Reproduction setup_**

OpenBind-0
- OpenFold3 version: v0.5.0
- Checkpoint: of3-ob-2025-06-30-174k.pt
- Training data cutoff: June 30, 2025

I evaluated two configurations:

1. OpenBind-0 without templates
- Released Zenodo query files and MSAs (uniref90_hits.sto, cfdb_hits.a3m and mgnify_hits.sto for unpaired MSA and uniprot_hits.sto for paired MSA)
- --use-msa-server false
- --use-templates false

2. OpenBind-0 with templates
- Released query, MSA, and template information (hmm_output.sto and local PDB mmCIF directory)
- --use-msa-server false
- --use-templates true
- Maximum template release date: 2025-06-30
- n_templates: 4
- max_templates: 20

OpenFold3-preview2 comparator
- OpenFold3 version: v0.4.5
- Checkpoint: of3-p2-155k.pt
- Training cutoff: September 2021

Shared inference settings
- Queries: 462
- Random seeds: 42, 43, 44, 45, 46
- Diffusion samples per seed: 5
- Total predictions per query: 25
- Recycles: 10
- PyTorch Lightning precision: 32-true
- OpenFold3 predict and low_mem presets
- cuEquivariance triangle kernels enabled
- Released MSAs were supplied directly; the MSA server was disabled
- Main MSA subsampling disabled
- Steering disabled
- GPU: A100-SXM-80GB

Evaluation protocol

I used the OST evaluation implementation from the official OpenFold3 benchmarking repository (https://github.com/aqlaboratory/openfold-3-benchmarking) to calculate lDDT-PLI and L-RMSD.

The two selection protocols were:

- Top-1: the sample with the highest average chain-pair ipTM over ligand-to-other-polymer pairs
- Oracle: any successful sample among the 25 predictions

----

So, here, Could you please provide, or point me to, the following information?
1. The complete end-to-end RnP2025-preview reproduction script used for Figures 1–5, starting from the released queries.json, MSAs, templates, and checkpoints.
2. The exact OpenFold3 git commit, environment, dependency versions, and inference configuration used for each OpenBind-0 and OpenFold3-preview2.
3. The PoseBusters version, configuration, and per-sample validity results used for Figure 2.
4. The supported steering functionality, inference command, and any additional configuration.
5. Confirmation of whether any query fixes, MSA/template preprocessing, template filtering, sample exclusions, or ranking post-processing were applied after the Zenodo release inputs were generated.

**Comment by HBioquant** [comment]
Here, I will briefly demonstrate an example of an input query and the configuration in config.yaml：

**input query of OpenBind-0 with templates**
```
{
  "seeds": [
    42, 43, 44, 45, 46
  ],
  "queries": {
    "10dj__1__1.A__1.C": {
      "chains": [
        {
          "chain_ids": [
            "A"
          ],
          "molecule_type": "protein",
          "sequence": "VWEIPRESLQLIKRLG",
          "main_msa_file_paths": [
            "/path/to/data/of3_MSA/rep_msas/p925babf455e1"
          ],
          "template_alignment_file_path": "/path/to/of3_MSA/rep_msas/p925babf455e1/hmm_output.sto"
        },
        {
          "chain_ids": [
            "C"
          ],
          "molecule_type": "ligand",
          "smiles": "CN1CCN(CCOc2cc(OC3CCOCC3)c3c(Nc4c(Cl)ccc5c4OCO5)ncnc3c2)CC1"
        }
      ]
    },
}
```
**config.yaml**
```
pl_trainer_args:
  precision: 32-true
experiment_settings:
  skip_existing: true
  seeds:
  - 42
  - 43
  - 44
  - 45
  - 46
model_update:
  presets:
  - predict
  - low_mem
  custom:
    settings:
      memory:
        eval:
          use_cueq_triangle_kernels: true
    architecture:
      shared:
        num_recycles: 10
      heads:
        pae:
          enabled: true
      msa:
        msa_module_embedder:
          subsample_main_msa: false
          subsample_all_msa: true
output_writer_settings:
  structure_format: cif
data_module_args:
  num_workers: 1
dataset_config_kwargs:
  msa:
    max_seq_counts:
      uniprot_hits: 50000
      mgnify_hits: 5000
      uniref90_hits: 10000
      bfd_hits: 100000000
      cfdb_hits: 10000000
      cfdb_uniref30: 10000000
      bfd_uniref_hits: 10000000
      rfam_hits: 10000
      rnacentral_hits: 10000
      nt_hits: 10000
      nucleotide_collection_hits: 10000
    msas_to_pair:
    - uniprot_hits
    aln_order:
    - uniref90_hits
    - bfd_hits
    - cfdb_hits
    - cfdb_uniref30
    - bfd_uniref_hits
    - mgnify_hits
    - rfam_hits
    - rnacentral_hits
    - nt_hits
    - nucleotide_collection_hits
    subsample_main: false
  template:
    n_templates: 4
    take_top_k: true
template_preprocessor_settings:
  max_release_date: '2025-06-30'
  min_release_date_diff: null
  max_sequences_parse: 200
  max_templates: 20
  structure_directory: /path/to/data/PDB/pdb_mmcif/mmcif_files
  structure_file_format: cif
  cache_directory: /path/to/data/of3_MSA/template_cache
  fetch_missing_structures: false
  preprocess_timeout: 600
```

**Comment by jandom** [comment]
Thank your for reporting @HBioquant , we've assigned @etowahadams to help with this

**Comment by etowahadams** [comment]
Hi @HBioquant,

Thanks for reporting this issue. I believe the discrepancy you're seeing is because of the use of OST, which (as you rightly inferred) is what we usually use for benchmarking. 

For these computations, we used PXMeter v1.1.1 which has its own implementation of lDDT-PLI and ligand RMSD, which are slightly different from OST. The reason is that initially we were benchmarking the OB0 model on the PXMeter2025H2 benchmarking set and using PXMeter infrastructure. But after we aquired the RnP style benchmark, we kept the PXMeter scoring infrastructure since it had built in functionality to aggregate the confidences from all the models we were working with. My apologies about any confusion 😅. 

For reference here is the `10dj__1__1.A__1.C` predictions scored with OST and PXM: 

metric | OST top-1 (ipTM) | PXMeter top-1 (ipTM) | OST oracle | PXMeter oracle
-- | -- | -- | -- | --
lDDT-PLI | 0.9640 | 0.9822 | 0.9664 | 0.9842
ligand RMSD (Å) | 0.398 | 0.380 | 0.341 | 0.328

Best,
Etowah
